tcms/xmlrpc/serializer.py

```diff
@@ -121,15 +121,58 @@ def serialize_queryset(self):
         Returns: List
         """
         response = []
-        for m in self.queryset:
-            self.model = m
-            m = self.serialize_model()
-            response.append(m)
+        for model in self.queryset:
+            self.model = model
+            model = self.serialize_model()
+            response.append(model)
 
         del self.queryset
         return response
 
 
+def _get_single_field_related_object_pks(m2m_field_query, model_pk, field_name):
+    field_names = []
+    for item in m2m_field_query[model_pk]:
+        if item[field_name]:
+            field_names.append(item[field_name])
+    return field_names
+
+
+def _get_related_object_pks(m2m_fields_query, model_pk, field_name):
+    """Return related object pks from query result via ManyToManyFields
+
+    Any object pk with value 0 or None values will be excluded in the final
+    list.
+
+    :param dict m2m_field_query: the result returned from _query_m2m_fields
+    :param model_pk: whose object's related object pks will be retrieved
+    :type model_pk: int or long
+    :param str field_name: field name of the related object
+    :return: list of related objects' pks
+    :rtype: list
+    """
+    data = m2m_fields_query[field_name]
+    return _get_single_field_related_object_pks(data, model_pk, field_name)
+
+
+def _serialize_names(row, values_fields_mapping):
+    # Replace name from ORM side to the serialization side as expected
+    new_serialized_data = {}
+    if values_fields_mapping:
+        for orm_name, serialize_info in \
+                values_fields_mapping.items():
+            serialize_name, conv_func = serialize_info
+            value = conv_func(row[orm_name])
+            new_serialized_data[serialize_name] = value
```

Comment:
You can reverse the logic so the more unlikely scenario is in the `if` case, e.g.
```
if not values_fields_mapping:
    new_serialized_data.update(row)
    return new_serialized_data
for orm_name, serialize_info ....
```